Not working on react native 0.71.12 on windows

- React Native ---> 0.71.12
- react-native-keys ---> 0.5.11
- OS ---> windows 10
- newArchEnabled ---> false

I simply create a `keys.development.json` in the root of my project with **secure** and **public** objects. then build the project with `npx react-native run-android` command. but I cannot get strings in my components neither with `Keys.secureFor('some secret')` nor `Keys.somePublicString`. I checked the example folder but it did not help me!

Can anyone tell me what I miss here?

I applied changes in `app/build.gradle`, clean project (`gradlew clean`), then build it, but ran into following error:

```
> Configure project :app



*********************************************************************************************************************
***                             If you are running build flavour then                                            ****
*** from app/build.gralde file, project.ext.keyFiles is missing key for debug flavour, please define it.
*********************************************************************************************************************



Choose keys.development.json file for react-native-keys to avoid stop any build process


5 actionable tasks: 5 up-to-date

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Script 'F:\PeakTower\Main projects\digivcard-app\node_modules\react-native-keys\android\RNKeys.gradle' line: 86

* What went wrong:
A problem occurred evaluating script.
> Cannot run program "/bin/sh": CreateProcess error=2, The system cannot find the file specified

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
A problem occurred configuring project ':app'.
> compileSdkVersion is not specified. Please add it to build.gradle

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 9s

error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup.
Error: Command failed: gradlew.bat app:installDebug -PreactNativeDevServerPort=8081

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Script 'F:\PeakTower\Main projects\digivcard-app\node_modules\react-native-keys\android\RNKeys.gradle' line: 86

* What went wrong:
A problem occurred evaluating script.
> Cannot run program "/bin/sh": CreateProcess error=2, The system cannot find the file specified

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
A problem occurred configuring project ':app'.
> compileSdkVersion is not specified. Please add it to build.gradle

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 9s

    at makeError (F:\PeakTower\Main projects\digivcard-app\node_modules\execa\index.js:174:9)
    at F:\PeakTower\Main projects\digivcard-app\node_modules\execa\index.js:278:16
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async runOnAllDevices (F:\PeakTower\Main projects\digivcard-app\node_modules\@react-native-community\cli-platform-android\build\commands\runAndroid\runOnAllDevices.js:82:7)
    at async Command.handleAction (F:\PeakTower\Main projects\digivcard-app\node_modules\@react-native-community\cli\build\index.js:108:9)
info Run CLI with --verbose flag for more details.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

--- a/android/RNKeys.gradle
+++ b/android/RNKeys.gradle
@@ -45,18 +45,34 @@ def loadKeys() {
     def exportCommand = "";
     if (System.env['KEYSFILE']) {
       keysFile = System.env['KEYSFILE']
-      exportCommand = "export KEYSFILE=" + keysFile + " && ";
+      if (System.properties['os.name'].toLowerCase().contains('windows')) {
+        exportCommand = "set KEYSFILE=" + keysFile + " && node ";
+      } else {
+        exportCommand = "export KEYSFILE=" + keysFile + " && ";
+      }
     } else if (System.env['DEFAULT_FILE_NAME']) {
       keysFile = System.env['DEFAULT_FILE_NAME']
-      exportCommand = "export KEYSFILE=" + keysFile + " && ";
+      if (System.properties['os.name'].toLowerCase().contains('windows')) {
+        exportCommand = "set KEYSFILE=" + keysFile + " && node ";
+      } else {
+        exportCommand = "export KEYSFILE=" + keysFile + " && ";
+      }
     } else if (project.hasProperty("DEFAULT_FILE_NAME")) {
       keysFile = project.ext.DEFAULT_FILE_NAME
-      exportCommand = "export KEYSFILE=" + keysFile + " && ";
+      if (System.properties['os.name'].toLowerCase().contains('windows')) {
+        exportCommand = "set KEYSFILE=" + keysFile + " && node ";
+      } else {
+        exportCommand = "export KEYSFILE=" + keysFile + " && ";
+      }
     } else if (project.hasProperty("keyFiles")) {
       project.ext.keyFiles.any { pair ->
           if (flavor.startsWith(pair.key.toLowerCase())) {
             keysFile = pair.value
-            exportCommand = "export KEYSFILE=" + keysFile + " && ";
+            if (System.properties['os.name'].toLowerCase().contains('windows')) {
+              exportCommand = "set KEYSFILE=" + keysFile + " && node ";
+            } else {
+              exportCommand = "export KEYSFILE=" + keysFile + " && ";
+            }
             return true
           }
       }
@@ -71,7 +87,11 @@ def loadKeys() {
       println("\n\n")
       if (project.ext.has('keyFiles') && project.ext.keyFiles.size() > 0) {
         keysFile = project.ext.keyFiles[project.ext.keyFiles.keySet().iterator().next()];
-        exportCommand = "export KEYSFILE=" + keysFile + " && ";
+        if (System.properties['os.name'].toLowerCase().contains('windows')) {
+          exportCommand = "set KEYSFILE=" + keysFile + " && node ";
+        } else {
+          exportCommand = "export KEYSFILE=" + keysFile + " && ";
+        }
         println "Choose " + keysFile + " file for react-native-keys to avoid stop any build process\n\n"
       } else {
         // Print an error message
@@ -88,7 +108,13 @@ def loadKeys() {
       def nodeModulesDir = getNodeModulesDir()
       exportCommand = exportCommand + nodeModulesDir + "/node_modules/react-native-keys/keysAndroid.js"
     }
-    def proc = ["/bin/sh", "-c", exportCommand].execute()
+    def proc;
+
+    if (System.properties['os.name'].toLowerCase().contains('windows')) {
+      proc = ["cmd", "/c", exportCommand].execute()
+    } else {
+      proc = ["/bin/sh", "-c", exportCommand].execute()
+    }
     proc.consumeProcessOutput(System.out, System.err)
 
     println(keysFile)

